### ISSUE TITLE
Old commits of the commits timeline is not updated with the main branch in pr page

### ISSUE THREAD

**Issue by hiifong** [go-gitea/gitea#25870]
### Description

Related: #25827

commits timeline of conversation tab have some problem in the pr page,this have a main branch in gitea repo,when merged upstream to the fix branch in local gitea repo,  push it to gitea fix branch,new a pull request from fix to main,then fetch upstream to main and merge, push it to gitea main branch,old commits of the commits timeline is not updated with the main branch.
![ca9af4788cf5f70f007256eb68c8e2d0](https://github.com/go-gitea/gitea/assets/89133723/9942f5cc-2ab0-4ea3-8675-8e7f4c765282)
![f13fe8f778564eb4b48c919ae14a4f27](https://github.com/go-gitea/gitea/assets/89133723/d43ddb19-f3c3-466f-b36e-e1ad12c16887)

### Gitea Version

v1.20+

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

./gitea web

### Database

MySQL

**Comment by silverwind** [comment]
Yes, I recall this asynchronity between timeline commits and commits list als happens when force-pushing the branch.

I think Gitea may incorrectly treat the timeline commits as append-only while in reality, it can always be rewritten.
